src/HttpAdapterDiscovery.php

```diff
@@ -11,19 +11,28 @@
 
 namespace Http\Discovery;
 
+use Http\Adapter\HttpAdapter;
+
 /**
  * Finds an HTTP Adapter
  *
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
  */
-class HttpAdapterDiscovery
+class HttpAdapterDiscovery extends ClassDiscovery
 {
     /**
      * @var array
      */
-    protected static $adapters = [
-        'guzzle6' => 'Http\Adapter\Guzzle6HttpAdapter',
-        'guzzle5' => 'Http\Adapter\Guzzle5HttpAdapter',
+    protected static $classes = [
+        'guzzle6' => [
+            'condition' => 'Http\Adapter\Guzzle6HttpAdapter',
+            'class' => 'Http\Adapter\Guzzle6HttpAdapter'
```

Comment:
This is duplicated, but I like that it is explicit about that it’s going to return the same class as it finds.